Change the way we switch connection when the current one is not working

hmusum opened the pull request:
Consider only healthy sources (if there are any) when switching to
a new one due to failures.
Also log when connecting to a source.

Commit:
Change the way we switch connection when the current one is not working

Consider only healthy sources (if there are any) when switching to
a new one fue to failures.
Also log when connecting to a source

hakonhall commented on config/src/main/java/com/yahoo/vespa/config/JRTConnectionPool.java:
```diff
@@ -67,20 +65,34 @@ public synchronized JRTConnection getCurrent() {
-        currentConnection = sources.get(ThreadLocalRandom.current().nextInt(0, sources.size()));
+        if (sources.size() <= 1) return currentConnection;
+
+        List<JRTConnection> healthySources = sources.stream()
```
This selects the healthy connections.  But isn't the healthiness of a connection only updated when it is "current"?  And so what may happen is:
* Say there are 3 members A, B, and C.  They start out as healthy, and A is chosen to be current.
* Say A becomes unhealhty, and then a random of B and C is chosen, say B.
* While B is the current connection, A has no chance of changing its healthiness so it stays unhealthy.
* When B becomes unhealthy, C is chosen.
* When C becomes unhealthy, C is kept per below test on `size() == 0`

hmusum replied:
Yes, you are right. We could either start doing ping out-of-band or try an unhealthy connection after some time (if there is not any healthy ones to choose from). Will look into it.

hakonhall replied:
setNewCurrentConnection() is only called on creation or when exactly 1 connection is bad.  So a simple method might be to select one randomly, or select randomly except that one connection, respectively?

hmusum replied:
Thanks, I have implemented a suggestion with some test

Commit:
Exclude current sources when all sources are unhealthy

When choosing a new source when all are unhealthy exclude the current
source.
Minor changes to toString for JRTConnection
